### PR TITLE
Show collection UI elements based on security

### PR DESCRIPTION
**Pull request by danrot**
Merge together with https://github.com/massiveart/husky/pull/558

__tasks:__

- [ ] test coverage
- [ ] gather feedback for my changes
- [ ] submit changes to the documentation
- [x] apply security changes immediately
- [x] remove add collection button if permission not granted
- [x] create translations

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | none
| BC breaks        | none
| Documentation PR | none

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/main.js [hunk 1]
```diff
@@ -184,7 +184,7 @@ define([
             // getter for content data
             this.sandbox.on('sulu.content.contents.get-data', function(callback) {
                 // deep copy of object
-                callback(JSON.parse(JSON.stringify(this.data)));
+                callback(this.sandbox.util.deepCopy(this.data));
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/main.js]: i would prefer using the `util` service here ...

File: src/Sulu/Component/Security/Serializer/Subscriber/SecuredEntitySubscriber.php [hunk 2]
```diff
@@ -11,22 +11,67 @@
...
+        }
+
+        if (!$object instanceof SecuredEntityInterface) {
+            return;
+        }
+
+        $event->getVisitor()->addData(
+            '_permissions',
+            $this->accessControlManager->getUserPermissions(
+                // TODO how to get locale
```
Comment (wachterjohannes) [inline src/Sulu/Component/Security/Serializer/Subscriber/SecuredEntitySubscriber.php]: why do you need the locale here?

**Comment by wachterjohannes** [comment]
@danrot changelog missing

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/main.js (hunk 1 above)
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/main.js]: Why?

File: src/Sulu/Component/Security/Serializer/Subscriber/SecuredEntitySubscriber.php (hunk 2 above)
Comment (danrot) [inline src/Sulu/Component/Security/Serializer/Subscriber/SecuredEntitySubscriber.php]: I think you are right, it's not needed, will remove it.

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/main.js (hunk 1 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/main.js]: i think its a clear direct dependency ... but ok (: we could discuss the service concept afterwards (:

**Comment by wachterjohannes** [comment]
i have created a new collection and toke me off the `add` permission. after the OK of the overlay the collection reloads and the "Add collection ´" button disappears. but after a complete refresh with `cmd + r` the add button is there and i can open the overlay (after creating a new one i get a not permitted exception - so thats ok).

**Comment by wachterjohannes** [comment]
if i has a collection which i´m not permitted to see - i see the medias of that collection!

**Comment by danrot** [comment]
Where do you see the medias? I cannot access the collection, the collection itself is not listed, and the media is also not showing up on the initial screen "all media".

**Comment by wachterjohannes** [comment]
ok i think i am wrong with the images in root (: but i have found another issue! if your are in a collection without add permission and you go back the add button is hidden (also if i am on the root).

**Comment by danrot** [comment]
@wachterjohannes This issue should now be solved.

**Pull request merged**